pkg/clock/clock.go

```diff
@@ -63,13 +63,10 @@ func Reset() *clockapi.Mock {
 // package.
 type Clock struct {
 	mock *clockapi.Mock
-	sync.Mutex
```

Comment:
The underlying mutex was causing issues with the linter when any `Session` objects were copied by value. This pattern was used heavily in tests, so it was a real hassle.

Since this is the mutex on the local Clock instance (not global) and didn't even apply to time operations, I'm not sure how exactly test code could footgun itself by running parallel Set/Add/Reset -- I don't see a pattern where on a localized Clock, it would somehow be parallized but test management code would be fast-forwarding a single local in parallel? Seems like an anti-pattern if it were tried 🤷‍♂️ 

And the underlying `clockapi.Mock` manages itself with locks throughout.